staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -914,36 +914,65 @@ func (a *jwtAuthenticator) getUID(ctx context.Context, c claims, claimsUnstructu
 	return evalResult.EvalResult.Value().(string), nil
 }
 
-func (a *jwtAuthenticator) getExtra(ctx context.Context, claimsUnstructured *unstructured.Unstructured) (map[string][]string, error) {
-	if a.celMapper.Extra == nil {
-		return nil, nil
-	}
+func (a *jwtAuthenticator) getExtra(ctx context.Context, c claims, claimsUnstructured *unstructured.Unstructured) (map[string][]string, error) {
+	extra := make(map[string][]string)
 
-	evalResult, err := a.celMapper.Extra.EvalClaimMappings(ctx, claimsUnstructured)
-	if err != nil {
+	if credentialID, err := getCredentialID(c); err != nil {
 		return nil, err
+	} else if len(credentialID) > 0 {
+		extra[user.CredentialIDKey] = []string{credentialID}
+	}
+
+	if a.celMapper.Extra != nil {
+		evalResult, err := a.celMapper.Extra.EvalClaimMappings(ctx, claimsUnstructured)
+		if err != nil {
+			return nil, err
+		}
+		if err := populateExtraFromEvalResult(evalResult, extra); err != nil {
+			return nil, err
+		}
+	}
+
+	return extra, nil
+}
+
+func getCredentialID(c claims) (string, error) {
+	if _, ok := c["jti"]; ok {
+		var jti string
+		if err := c.unmarshalClaim("jti", &jti); err != nil {
+			return "", fmt.Errorf("oidc: parse jti claim: %w", err)
+		}
+
+		return credentialIDForJTI(jti), nil
+	}
+
+	return "", nil
+}
+
+func credentialIDForJTI(jti string) string {
```

Comment:
not a big deal, but consider relocating the existing `CredentialIDForJTI` function to a place both service account and oidc authenticators can call it (staging/src/k8s.io/apiserver/pkg/authentication/token/jwt, etc)